core/domain/base_model_validators.py

```diff
@@ -97,9 +98,17 @@ def __init__(self, field_name, model_class, model_ids):
                 'When fetching instances of UserSettingsModel, please use ' +
                 'UserSettingsModelFetcherDetails instead of ' +
                 'ExternalModelFetcherDetails')
+        filtered_model_ids = []
+        for model_id in model_ids:
+            if not model_id:
+                raise Exception(
+                    'A model id in the field \'%s\' '
+                    'is empty' % field_name)
+            else:
+                filtered_model_ids.append(model_id)
```

Comment:
Do we really need to filter the model ids, I mean if any of the ids are not in the expected pattern the cheek will fail, right?

If yes, then should we change the check to this:
```
if not all(model_ids):
  raise Exception(...) 
```

Reply:
Done.

Reply:
Hmm, I actually think that it is fine to filter just some models because if we have a long list of IDs and only one is bad, we probably want to try to fetch the rest and have all the results in one run of the job. If it failed even on one bad ID we would have to fix that separately and then rerun the job again to see the other errors.

Reply:
Ok then I have reverted it back how it was, I have added an else part.

Reply:
>  if we have a long list of IDs and only one is bad, we probably want to try to fetch the rest and have all the results in one run of the job.

@vojtechjelinek As we are raising validation error for any of the bad IDs, I think the function won't work for all other ids in the list and the job will stop there, right?

Reply:
Right, didn't realize this, I guess it depends on how hard it would be to implement this in a way that would report wrong IDs and also fetch the good ones if it is to complicated we can go by the approach that @DubeySandeep suggested initially.

Reply:
@vojtechjelinek  If we move the user validation to _fetch_field_name_to_external_model_references method i.e before _get_external_id_relationships is being called, we can add an error without using an exception, 

Another solution would be to add another field in ExternalModelFetcherDetails that contains the invalid ids, which could then be checked in _fetch_field_name_to_external_model_references method and add the errors. 

By doing it in the above ways, the job will not stop for this particular model.

Reply:
> Another solution would be to add another field in ExternalModelFetcherDetails that contains the invalid ids, which could then be checked in _fetch_field_name_to_external_model_references method and add the errors.

This seems like a good approach, but maybe it would be better to not only record the wrong IDs but record the errors directly also in this model and then pass them to the errors in the validator


I'm also fine with the first proposed approach. The decision is up to you.

Reply:
I have added a field to the class for model id errors. Done.